java/memory/src/main/java/org/apache/arrow/memory/BaseAllocator.java

```diff
@@ -237,14 +239,25 @@ private ArrowBuf bufferWithoutReservation(final int size, BufferManager bufferMa
 
     final AllocationManager manager = new AllocationManager(this, size);
     final BufferLedger ledger = manager.associate(this); // +1 ref cnt (required)
-    final ArrowBuf buffer = ledger.newArrowBuf(0, size, bufferManager);
+    
+    int offset = calcOffset(ledger.getUnderlying());
+    final ArrowBuf buffer = ledger.newArrowBuf(offset, size, bufferManager);
```

Comment:
it looks like we are relying on netty's behavior to allocate a size that is a multiple of CACHE_ALIGN
since we're changing offset without worrying about offset + size being greater than the underlying buffer.
Should we explicitly allocate a size that is a multiple of CACHE_ALIGN?
for example:
```
int sizePadding = size % CACHE_ALIGN == 0 ? 0 : CACHE_ALIGN - deltaToAligned;
final AllocationManager manager = new AllocationManager(this, size + sizePadding);
```

Even this does not seem to guarantee that the actual size will be big enough for the offset.
Am I missing something?